src/main/java/org/elasticsearch/search/aggregations/pipeline/movavg/MovAvgPipelineAggregator.java

```diff
@@ -158,13 +157,35 @@ public InternalAggregation reduce(InternalAggregation aggregation, ReduceContext
 
             double[] predictions = model.predict(values, predict);
             for (int i = 0; i < predictions.length; i++) {
```

Comment:
No particularly compelling reason, just how I was thinking about it at the time.  E.g. the first prediction is one ahead of the last valid bucket, second prediction is two ahead, etc.

Can switch it if you think it is cleaner.  Will need to change `predictions[i]` to `predictions[i - 1]` to compensate.